addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -150,13 +150,20 @@ odoo.define('website_form.s_website_form', function (require) {
             // force server date format usage for existing fields
             this.$target.find('.s_website_form_field:not(.s_website_form_custom)')
             .find('.s_website_form_date, .s_website_form_datetime').each(function () {
+                const $input = $(this).find('input');
+
+                // Datetimepicker('viewDate') will return `new Date()` 
+                // if the input is empty but we want to keep the empty value
```

Comment:
`if the` can be moved the comment line above, take all the space up to 80, don't return before for no reason, see https://github.com/odoo/odoo/pull/83508#discussion_r796442104